java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -172,18 +172,19 @@ private static boolean preConditionChecker(HintContext ctx) {
         }
 
         TreePath treePath = ctx.getPath();
-        EnhancedForLoopTree efl = null;
-        TypeMirror expressionType = null;
         if (treePath.getLeaf().getKind() == Tree.Kind.ENHANCED_FOR_LOOP) {
-            efl = (EnhancedForLoopTree) treePath.getLeaf();
-            expressionType = ctx.getInfo().getTrees().getTypeMirror(new TreePath(treePath, efl.getExpression()));
-        }
-
-        Element treePathElement = info.getTrees().getElement(treePath);
-        // variable should have local scope
-        //No need to DeclaredType check as primitive types can also be replaced with var in enhanced-for-loop
-        if ((treePathElement != null && treePathElement.getKind() != ElementKind.LOCAL_VARIABLE) && expressionType == null) {
-            return false;
+            EnhancedForLoopTree efl = (EnhancedForLoopTree) treePath.getLeaf();
+            TypeMirror expressionType = ctx.getInfo().getTrees().getTypeMirror(new TreePath(treePath, efl.getExpression()));
+            //No need to DeclaredType check as primitive types can also be replaced with var in enhanced-for-loop
+            if (expressionType == null) {
```

Comment:
I would prefer `Utilities.isValidType()`; unless the code is prepared to deal with incomplete and erroneous types later.

Reply:
We are checking for ```Utilities.isValidType()``` later in the code, but there is no harm in doing that check here as well, as it checks for null value too. Added that in the latest commit.